### DISCUSSION TITLE
[SPACESHIP] Create Sandbox Tester

### DISCUSSION THREAD

**Discussion by mateuszczechowski** [fastlane/fastlane#18252] [Q&A]
Hi!

Is it possible to use App Store Connect API KEY to create Sandbox Tester ?

I've checked docs: https://rdoc.info/gems/fastlane/Spaceship%2FConnectAPI%2FSandboxTester%2Ecreate 
but i cannot find information how to use API Key.

Thanks in advance

**Comment by dinsen** [comment]
Hi @mateuszczechowski 

It dosen't look like there is an endpoint for adding Sandbox Testers with the App Store Connect API. 

And I can see that when the code was added, it use the normal login:
https://github.com/fastlane/fastlane/pull/16742

**Reply by mateuszczechowski** [reply]
Hi @Dinsen 

Thank you for your replay! 

Yes, I am using it with normal login but as we needed to turn on 2FA our solution with normal login doesnt work.
We were trying using Application Specific Password and Fastlane Session for this and it works but fastlane session needs to be regenerated every 8h on our Ci. Do you think there is any other solution for this ?

**Reply by dinsen** [reply]
You can't use App Specific Password since this only works with iTMSTransporter (Upload .ipa to App Store)

Try look in this discussion, some users don't have that problem. Maybe some of the answers can help you:
https://github.com/fastlane/fastlane/discussions/17485

**Reply by jayantsarda** [reply]
Can someone suggest what does it take to implement it with API key ? Seems like https://github.com/fastlane/fastlane/discussions/17485 is already answered and should be solved now
